### PR TITLE
Fixes: testing form that consumes dataset

### PR DESCRIPTION
**Pull request by sadiqkhoja**
# Changes
- added dataset download ability to testing endpoint
- moved the common logic to query module

#### What has been done to verify that this works as intended?

Integration test + verified running it locally

#### Why is this the best possible solution? Were any other approaches considered?

I could have kept the common logic in resource module but that would entail passing 6+ parameters to new function because we container is only injected in query module.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

No

#### Does this change require updates to the API documentation? If so, please update docs/api.md as part of this PR.

No

#### Before submitting this PR, please make sure you have:

- [x] run `make test-full` and confirmed all checks still pass OR confirm CircleCI build passes
- [x] verified that any code from external sources are properly credited in comments or that everything is internally sourced

### CODE REVIEW

File: lib/model/query/form-attachments.js [hunk 1]
```diff
@@ -141,10 +144,32 @@ select ${_unjoinMd5.fields} from form_attachments
   where "formDefId"=${formDefId}`)
   .then(map(_unjoinMd5));
 
+const streamAttachment = (attachment, response) => async ({ Blobs, Datasets, Entities }) => {
+  if (attachment.blobId == null && attachment.datasetId == null) {
+    return reject(Problem.user.notFound());
+  } else if (attachment.blobId != null) {
+    const blob = await Blobs.getById(attachment.blobId).then(getOrNotFound);
+    return withEtag(`"${blob.md5}"`, () => binary(blob.contentType, attachment.name, blob.content));
```
Comment (matthew-white) [inline lib/model/query/form-attachments.js:152]: Doesn't `withEtag()` add its own quotes now? Should these quotes be removed?

**Review by matthew-white — APPROVED** [review]

**Comment by matthew-white** [comment]
> I could have kept the common logic in resource module but that would entail passing 6+ parameters to new function

Could you pass the function the container that's passed to the endpoint? Something like:

```js
service.get(`${base}/attachments/:name`, endpoint((container, { params, auth }, request, response) => {
  const { FormAttachments, Forms } = container;
  return getInstance(Forms, params)
    .then((form) => auth.canOrReject('form.read', form))
    .then((form) => FormAttachments.getByFormDefIdAndName(form.def.id, params.name)
      .then(getOrNotFound)
      .then(attachment => streamAttachment(container, attachment, response)));
}));
```

Not a huge deal from my perspective (I'll let you decide what's best), but I do think it's useful in general to keep the resource/query layers distinct.

**Comment by matthew-white** [comment]
@sadiqkhoja, do you recommend that the QA team verify this change? If so, could you add the "needs testing" label to this PR, as well as any notes for the QA team?

**Comment by sadiqkhoja** [comment]
# Steps to test:

- Create a draft Form that consumes a dataset
- Without publishing the form, go to "Testing" tab and click new
- Form should be opened in Enketo without any error
- Test the same form in ODK collect as well

**Comment by dbemke** [comment]
Tested with success!

**Pull request merged**